app/src/main/java/com/awscherb/cardkeeper/data/handler/BaseRealmHandler.java

```diff
@@ -84,16 +67,67 @@
     //================================================================================
 
     @Override
-    public Observable<Void> deleteObject(T object) {
+    public Completable deleteObject(T object) {
         Realm realm = Realm.getDefaultInstance();
         return doAsync(object, realm, o ->
                 realm.where(getModelClass())
                         .equalTo("id", o.getId())
                         .findAll()
                         .first()
-                        .deleteFromRealm())
-                .map(delete -> null);
+                        .deleteFromRealm()).toCompletable();
+    }
+
+    //================================================================================
+    // Helper
+    //================================================================================
+
+    private Single<T> doAsync(T object, Realm realm, Consumer<T> consumer) {
+        return Single.create(e -> {
```

Comment:
Yes, `create` is safe in RxJava 2, for Single you don't even have to worry about backpressure strategy.